dare.go

```diff
@@ -44,9 +44,9 @@ const (
 
 var (
 	errMissingHeader      = errors.New("dare: incomplete header")
```

Comment:
We have to think about the name... AEAD is a cryptographic term for a class/type of algorithms. Can confuse users - e.g. we use AEADs in minio/aead -> recursion :wink: 

I used the simplest name I could come up with: Data At Rest Encryption -> DARE.
We are aware of the current inconstancy of the naming...